lib/members_page.rb

```diff
@@ -0,0 +1,70 @@
+# encoding: utf-8
+# frozen_string_literal: true
+require 'scraped'
+require_relative './decorators/party_name_attributes'
+
+class MembersPage < Scraped::HTML
+  decorator Scraped::Response::Decorator::AbsoluteUrls
+  decorator PartyNameAttributes
+
+  field :party_groupings do
+    noko.xpath('//td/table/tr/td[contains(.,"Circunscripción")]/parent::tr')
+        .map do |row|
+      fragment row => PartyGroupings
+    end
+  end
+
+  class PartyGroupings < Scraped::HTML
+    field :party do
+      party_image_node[:party_name]
+    end
+
+    field :party_id do
+      party_image_node[:party_id]
+    end
+
+    field :members do
```

Comment:
I think this one needs a comment to explain what's going on. I'm pretty sure there's going to be a better way to achieve this, but I can't even really tell what it's doing.